From ac5cc4a413bbb10ad03a3615f4fe718b805784de Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 21 May 2021 16:11:15 +0800
Subject: [PATCH] =?UTF-8?q?fix=20=E2=80=98numeric=5Flimits=E2=80=99=20is?=
 =?UTF-8?q?=20not=20a=20member=20of=20=E2=80=98std=E2=80=99?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Compling with GCC11, there is a failure
...
|external/com_google_absl/absl/synchronization/internal/graphcycles.cc:451:26:
error: 'numeric_limits' is not a member of 'std'
451 |   if (x->version == std::numeric_limits<uint32_t>::max()) {
...

Add #include <limits> to the affected file

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 ...m_google_absl_fix_mac_and_nvcc_build.patch | 44 +++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/third_party/com_google_absl_fix_mac_and_nvcc_build.patch b/third_party/com_google_absl_fix_mac_and_nvcc_build.patch
index 6301119ab2c..9b677b6c789 100644
--- a/third_party/com_google_absl_fix_mac_and_nvcc_build.patch
+++ b/third_party/com_google_absl_fix_mac_and_nvcc_build.patch
@@ -1,3 +1,32 @@
+From 7bbc3215c22a0d76a1755f9004c9bec490524f8c Mon Sep 17 00:00:00 2001
+From: Hongxu Jia <hongxu.jia@windriver.com>
+Date: Fri, 21 May 2021 16:07:26 +0800
+Subject: [PATCH] com google absl fix mac and nvcc build
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+fix ‘numeric_limits’ is not a member of ‘std’
+
+Compling with GCC11, there is a failure
+...
+|external/com_google_absl/absl/synchronization/internal/graphcycles.cc:451:26:
+error: 'numeric_limits' is not a member of 'std'
+451 |   if (x->version == std::numeric_limits<uint32_t>::max()) {
+...
+
+Add #include <limits> to the affected file
+
+Upstream-Status: Pending
+
+Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
+---
+ absl/container/internal/compressed_tuple.h   | 126 ++++---------------
+ absl/strings/string_view.h                   |  35 ++++++
+ absl/synchronization/internal/graphcycles.cc |   1 +
+ absl/time/internal/cctz/BUILD.bazel          |   9 --
+ 4 files changed, 63 insertions(+), 108 deletions(-)
+
 diff --git a/absl/container/internal/compressed_tuple.h b/absl/container/internal/compressed_tuple.h
 index 4bfe92f..01db713 100644
 --- a/absl/container/internal/compressed_tuple.h
@@ -288,6 +317,18 @@ index 1861ea6..c7a916b 100644
    }
  
    static constexpr size_type StrlenInternal(const char* str) {
+diff --git a/absl/synchronization/internal/graphcycles.cc b/absl/synchronization/internal/graphcycles.cc
+index 6a2bcdf..71a5d7d 100644
+--- a/absl/synchronization/internal/graphcycles.cc
++++ b/absl/synchronization/internal/graphcycles.cc
+@@ -36,6 +36,7 @@
+ #include "absl/synchronization/internal/graphcycles.h"
+ 
+ #include <algorithm>
++#include <limits>
+ #include <array>
+ #include "absl/base/internal/hide_ptr.h"
+ #include "absl/base/internal/raw_logging.h"
 diff --git a/absl/time/internal/cctz/BUILD.bazel b/absl/time/internal/cctz/BUILD.bazel
 index 7a53c81..159b0f0 100644
 --- a/absl/time/internal/cctz/BUILD.bazel
@@ -308,3 +349,6 @@ index 7a53c81..159b0f0 100644
      visibility = ["//visibility:public"],
      deps = [
          ":civil_time",
+-- 
+2.27.0
+
-- 
2.27.0

